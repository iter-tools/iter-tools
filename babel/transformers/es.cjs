const { resolve } = require('path');
const babelJest = require('babel-jest');

module.exports = babelJest.createTransformer({
  configFile: resolve(__dirname, '../jest.config.cjs'),
});
