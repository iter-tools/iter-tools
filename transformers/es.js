const createTransformer = require('./create-transformer');

module.exports = createTransformer('babel-es.config.js');
