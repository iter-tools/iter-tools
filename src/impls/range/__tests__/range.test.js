import { range } from '@iter-tools/es';
import { anyType, unwrap } from '../../../test/helpers.js';

describe('range', () => {
  describe('when no arguments are passed', () => {
    it('yields incrementing integers starting with 0', () => {
      const [a, b, c] = range();
      expect([a, b, c]).toEqual([0, 1, 2]);

      const [d, e, f] = range({});
      expect([d, e, f]).toEqual([0, 1, 2]);
    });
  });

  describe('when end is passed', () => {
    it('yields incrementing integers from 0 to end', () => {
      expect(unwrap(range(3))).toEqual([0, 1, 2]);
      expect(unwrap(range({ end: 3 }))).toEqual([0, 1, 2]);
    });
  });

  describe('when start and end are passed', () => {
    it('yields incrementing integers from start to end', () => {
      expect(unwrap(range(1, 4))).toEqual([1, 2, 3]);
      expect(unwrap(range({ start: 1, end: 4 }))).toEqual([1, 2, 3]);
    });
  });

  describe('when start, end, and step are passed', () => {
    it('yields integers from start to end incrementing by step', () => {
      expect(unwrap(range(1, 7, 2))).toEqual([1, 3, 5]);
      expect(unwrap(range(1, 6, 2))).toEqual([1, 3, 5]);
      expect(unwrap(range({ start: 1, end: 6, step: 2 }))).toEqual([1, 3, 5]);
    });
  });

  describe('when options are invalid', () => {
    it('throws', () => {
      expect(() => range({ start: anyType('foo') })).toThrowErrorMatchingSnapshot();

      expect(() => range({ end: anyType(null) })).toThrowErrorMatchingSnapshot();

      expect(() => range({ step: anyType(Infinity) })).toThrowErrorMatchingSnapshot();
    });
  });
});
