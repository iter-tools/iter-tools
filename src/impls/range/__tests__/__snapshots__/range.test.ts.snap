// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`range when options are invalid throws 1`] = `"The specified start was not an integer"`;

exports[`range when options are invalid throws 2`] = `"The specified end was not an integer or infinite"`;

exports[`range when options are invalid throws 3`] = `"The specified step was not a non-zero integer"`;
